map_view/show-location-history/src/main/java/com/esri/samples/show_location_history/ShowLocationHistorySample.java

```diff
@@ -0,0 +1,201 @@
+        SimpleLineSymbol.Style.SOLID, 0xFF00FF00, 2.0f);
+      SimpleRenderer locationHistoryLineRenderer = new SimpleRenderer(locationLineSymbol);
+      locationHistoryLineOverlay.setRenderer(locationHistoryLineRenderer);
+
+      // add the graphics overlays to the map view
+      mapView.getGraphicsOverlays().addAll(Arrays.asList(locationHistoryOverlay, locationHistoryLineOverlay));
+
+      // create a polyline builder to connect the location points
+      PolylineBuilder polylineBuilder = new PolylineBuilder(SpatialReference.create(3857));
+      // access the json of the location points
+      String polylineData = IOUtils.toString(getClass().getResourceAsStream("/show_location_history/polyline_data.json"),
+        StandardCharsets.UTF_8);
+      // create a polyline from the location points
+      Polyline routePolyline = (Polyline) Polyline.fromJson(polylineData,
+        SpatialReference.create(3857));
+
+      // create a simulated location data source
+      SimulatedLocationDataSource simulatedLocationDataSource = new SimulatedLocationDataSource();
+      // set the location of the simulated location data source with simulation parameters to set a consistent velocity
+      simulatedLocationDataSource.setLocations(
+        routePolyline, new SimulationParameters(Calendar.getInstance(), 30.0, 0.0, 0.0));
+
+      // start the simulated location data source
+      simulatedLocationDataSource.startAsync();
+      simulatedLocationDataSource.addLocationChangedListener(locationChangedEvent -> {
+        // if location tracking is turned off, do not add to the polyline
+        if (!isTrackingEnabled) {
+          return;
+        }
+        // get the position as a point from locationChangedEvent
+        Point position = locationChangedEvent.getLocation().getPosition();
+
+        // add the new point to the polyline
+        polylineBuilder.addPoint(position);
+
+        // add the new point to the graphics overlay
+        locationHistoryOverlay.getGraphics().add(new Graphic(position));
+
+        // reset the old polyline connecting the points
+        locationHistoryLineOverlay.getGraphics().clear();
+
+        // add the updated polyline to the graphics overlay
+        locationHistoryLineOverlay.getGraphics().add(new Graphic(polylineBuilder.toGeometry()));
+      });
+
+      // configure the map view's location display to follow the simulated location data source
+      mapView.getLocationDisplay().setLocationDataSource(simulatedLocationDataSource);
+      mapView.getLocationDisplay().setAutoPanMode(LocationDisplay.AutoPanMode.RECENTER);
+      mapView.getLocationDisplay().setInitialZoomScale(7000);
+
+      trackingButton.setOnAction(event -> {
+        // if the user has panned away from the location display, turn it on again
+        if (mapView.getLocationDisplay().getAutoPanMode() == LocationDisplay.AutoPanMode.OFF) {
+          mapView.getLocationDisplay().setAutoPanMode(LocationDisplay.AutoPanMode.RECENTER);
+        }
```

Comment:
I find the interaction with panning and switching the auto-pan on and off is not that intuitive? It doesn't help that the auto-pan functionality is not continuous, but only happens when the dot is approaching the edge of the screen. Not sure if this can be adjusted somehow.

 Perhaps this sample could benefit from a tick-box to enable/disable auto-pan? When the tick-box would be checked, the manual panning with the mouse should be disabled (or, panning with the mouse would de-activate that tickbox). What do you think?

Reply:
I agree, I think a checkbox toggling the auto-pan would be more intuitive. 

However, when the auto-pan is off the issue with the extra polylines still occurs when the user pans away.

Reply:
I'll have a closer look at the code. I didn't notice any connection between the drawing and listening to panning on the map, but it's possible that one thing is triggering another?

Reply:
It happens when `position = mapView.getLocationDisplay().getMapLocation()` so that the LocationDisplay symbol moves ahead of the tracking. When the position is set this way and auto-pan is disabled, if the user pans away then back to the tracking.

![image](https://user-images.githubusercontent.com/67917108/93580451-b6c58f80-f997-11ea-900c-3b7e9a478ce4.png)

Reply:
So I think this fixes the issue with the extra polylines, and also sets the `LocationDisplay` marker ahead of the tracking graphics:
```
// get the position as a point from the simulated location data source
Integer nextPoint = simulatedLocationDataSource.getCurrentLocationIndex()-1;
LocationDataSource.Location nextLocation = simulatedLocationDataSource.getLocations().get(nextPoint);
Point position = new Point(nextLocation.getPosition().getX(),nextLocation.getPosition().getY());
```

Reply:
I was thinking of something similar, but that feels a little counterintuitive from a user perspective.
I'd check whether any of the other platforms are doing something similar. If they aren't specifically working around this, then we may have a small bug in the LocationDisplay API on Java.